lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
+          unless what.is_a? String
+            if klass.to_s.classify.constantize.find_by_id(what.id)
+              what = what.to_label
+            else
+              what = what.class.name
```

Comment:
@ohadlevy this is to remove the labels of objects that aren't in the users taxonomy, is this what you wanted or do you prefer a count of how many associations the record has?